sched/scheduler/workGenerator_test.go

```diff
@@ -0,0 +1,54 @@
+package scheduler
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/queue"
+	"testing"
+)
+
+func TestGenerateWork_ReadsAllItemsFromChannel(t *testing.T) {
```

Comment:
I think this file would be a lot simpler if you just use a fake queue (the in-memory, simple one) instead of mocking.